src/__tests__/posthog-core.ts

```diff
@@ -387,6 +401,7 @@ describe('posthog core', () => {
                 properties: () => ({ distinct_id: 'abc', persistent: 'prop', $is_identified: false }),
                 remove_event_timer: jest.fn(),
                 get_property: () => 'anonymous',
+                props: { $groups: {} },
```

Comment:
adding this made most the tests pass with the new `identified_only` default config. Does this indicate that something else needs to happen somewhere to make sure real users always have this in their props? Or is it just a test fixture thing? (I imagine the latter because people can use `identifed_only` with no problem, but want to be certain)

Reply:
It feels like the latter, but I'll stare at this code a little to convince myself a bit more

Reply:
Ok, it was the absence of `props` in the mock that made the test fail, it didn't actually need `$groups: {}`, so I've slightly changed this line to have a more minimal mock